bootstrapvz/plugins/admin_user/tasks.py

```diff
@@ -42,13 +43,62 @@ def run(cls, info):
 
 
 class AdminUserCredentials(Task):
-	description = 'Modifying ec2-get-credentials to copy the ssh public key to the admin user'
+	description = 'Set up access credentials for the admin user'
 	phase = phases.system_modification
-	predecessors = [InstallInitScripts]
+	predecessors = [InstallInitScripts, CreateAdminUser]
 
 	@classmethod
 	def run(cls, info):
-		from bootstrapvz.common.tools import sed_i
-		getcreds_path = os.path.join(info.root, 'etc/init.d/ec2-get-credentials')
-		username = info.manifest.plugins['admin_user']['username']
-		sed_i(getcreds_path, 'username=\'root\'', 'username=\'{username}\''.format(username=username))
+                from bootstrapvz.common.exceptions import TaskError
+                from bootstrapvz.common.tools import sed_i
+                from bootstrapvz.common.tools import log_check_call
+                log = logging.getLogger(__name__)
+
+                if 'password' in info.manifest.plugins['admin_user']:
+                        if 'pubkey' in info.manifest.plugins['admin_user']:
+                                msg = 'The options password and pubkey are mutually exclusive'
+                                raise TaskError(msg)
+                        log_check_call(['chroot', info.root, 'chpasswd'],
+                                       info.manifest.plugins['admin_user']['username'] +
+                                       ':' + info.manifest.plugins['admin_user']['password'])
+                        return
+
+                getcreds_path = os.path.join(info.root, 'etc/init.d/ec2-get-credentials')
+                if 'pubkey' in info.manifest.plugins['admin_user']:
+                        import stat
+                        from shutil import copy
+                        full_path = info.manifest.plugins['admin_user']['pubkey']
+                        if not os.path.exists(full_path):
+                                msg = 'Could not find public key at {full_path}'.format(full_path=full_path)
+                                raise TaskError(msg)
+                        log.debug('Copying public key from {path}'.format(path=full_path))
+
+                        if os.path.exists(getcreds_path):
+                                log.warn('You are using a static public key for the admin account.'
+                                         ' This will conflict with the ec2 public key njection mechanisn.'
+                                         ' The ec2-get-credentials startup script has therefore been disabled.')
+                                log_check_call(['chroot', info.root, 'insserv', '--remove',
+                                                'ec2-get-credentials'])
+                        username = info.manifest.plugins['admin_user']['username']
+
+                        ssh_file = os.path.join('/home/'
```

Comment:
heh, you are even using `os.path.join()` mate. How about: `os.path.join('/home', username, '.ssh/authorized_keys')`? :-D